core/types/block_ext_test.go

```diff
@@ -112,108 +115,194 @@ func assertDifferentPointers[T any](t *testing.T, a *T, b any) {
 	// the memory address would be different as well.
 }
 
-func TestBlockExtraGetWith(t *testing.T) {
+func TestBodyRLP(t *testing.T) {
```

Comment:
Most of this is now a duplicate of `TestBodyExtraRLP()`.

There only needs to be `TestBodyRLP()` and `TestBlockRLP()` that both perform (a) RLP round-trip recovery; and (b) comparison to golden data.

While it's feasible to split them both into (a) and (b) components I think that adds unnecessary boilerplate when the golden-data comparison adds literally 2 lines (const and assertion).

Reply:
Done 👍